library-app/src/components/Sidebar/Sidebar.tsx

```diff
@@ -19,38 +23,40 @@ const Sidebar = ({ isLoggedIn }: MainLayoutProps) => {
     <React.Fragment>
       <nav className={styles.header}>
         <div className={styles['btn-holder']} onClick={handleHomeNavigation}>
-          <a>
+          <button>
             <img src={home} alt='' />
-          </a>
+          </button>
         </div>
         {isLoggedIn && (
           <div className={styles['btn-holder']}>
-            <a>
+            <button>
               <img src={account} alt='' />
-            </a>
+            </button>
           </div>
         )}
         {isLoggedIn && (
           <div className={styles['btn-holder']} onClick={() => setAdminOptions(!adminOptions)}>
-            <a>
+            <button>
               <img src={showMore} alt='' />
-            </a>
+            </button>
           </div>
         )}
         {adminOptions && (
           <nav className={styles.sidebar}>
             <div className={styles['btn-holder']}>
-              <a>
-                <img src={account} alt='admin option1' />
-              </a>
-            </div>
-            <div className={styles['btn-holder']}>
-              <a>
-                <img src={account} alt='admin option2' />
-              </a>
+              <button
+                className={styles['add-new-book']}
+                type='submit'
+                onClick={() => setIsOpen(true)}
+              >
+                Add New Book +
+              </button>
             </div>
           </nav>
         )}
+        <Modal open={isOpen} onClose={() => setIsOpen(false)}>
```

Comment:
See? You don't need to pass isOpen, you can simply add condition here and display it or not based on that condition